app/models/domain.rb

```diff
@@ -51,7 +51,7 @@ class Domain < ApplicationRecord
 
   set_crud_hooks :domain
 
-  apipie :class, desc: "A class representing #{model_name.human} object" do
+  apipie :class, desc: "A class representing #{model_name.singular.humanize(capitalize: false)} object" do
```

Comment:
@ofedoren I wonder if it makes sense to enhance apipie to provide a `desc_from_class` or some other way to automatically derive this.

Reply:
It seems like all the descriptions are quite the same, so yeah, it makes sense to add something like `default_desc`.  Since I've noticed that `apipie-dsl` needs some love, I'm going to add this as well.

Since, ideally we want to support translation for all the strings (and apipie-dsl doesn't have dependency on gettext), we shouldn't hardcode such string in the lib, but somehow provide it to be evaluated in the scope of a certain class. My suggestion is to add a new entry called e.g. `default_model_description` to the `config` (https://github.com/theforeman/foreman/blob/develop/config/initializers/apipie.rb#L3), which will accept a proc, which will be evaluated in the class (to be able to call `model_name`) and save the string in the object. This could be translated later via callback we provide in the same `config`. It would look like:
```ruby
ApipieDSL.configure do |config|
# ...
  config.default_model_description = ->(model) { N_("A class representing %s object") % model.model_name.singular.humanize(capitalize: false) }
# ...
end
```

That will be applied to all documented models unless there is an explicit override, such as `:desc` option or `short` method is called within a block. WDYT?

Reply:
You do provide a translate method as `ApipieDSL.translate(str, locale)`:

https://github.com/Apipie/apipie-dsl/blob/f7118ed268449199272ce1245b3e30aece84c7f8/lib/apipie_dsl/application.rb#L214-L220

AFAIK it doesn't support interpolating the translated strings with variables, which makes it rather hard to use. I think that would still be needed.

In the context of Foreman: do we configure the translate method:

https://github.com/theforeman/foreman/blob/d8200eeae33ff730cf85fcae9c47e92466c64a6a/config/initializers/apipie.rb#L22-L28

And I think we should rewrite that to use `FastGettext.with_locale`:
```ruby
config.translate = lambda do |str, loc|
  str ? FastGettext.with_locale(loc) { _(str) } : nil
end
```

So I think your suggestion makes sense, but I'm not sure how to hook it up properly.